shell/src/context_listener.rs

```diff
@@ -216,6 +216,13 @@ fn listen_protocol_events(
                 }
 
                 perform_context_action(&action, context)?;
+                // below logic should be driven by dedicated ContextAction events
```

Comment:
@mat13mn
is it possible to trigger this block_applied/cycle_started logic just for MarkMoveInMem/MarkSweepInMem??

Reply:
for backends other then markmove/mark sweep it does nothing

Reply:
This is a good candidate for a commit-event + subscribers logic we discussed earlier today. Maybe not viable as part of this PR, but please add a TODO with TE-440 (which I just created for that task, still have to fill the description).